test/StreamableGeneratorTest.cpp

```diff
@@ -74,3 +75,49 @@ TEST(StreamableGeneratorTest, TestGeneratorNextThrowsExceptionWhenDone) {
     ASSERT_STREQ(e.what(), "Coroutine is not active");
   }
 }
+
+std::string decompressData(std::string_view compressedData,
+                           auto&& decompressionFilter) {
+  std::string result;
+  io::filtering_ostream filterStream;
+  filterStream.push(decompressionFilter);
+  filterStream.push(io::back_inserter(result));
+
+  filterStream.write(compressedData.data(),
+                     static_cast<std::streamsize>(compressedData.size()));
+  return result;
+}
+
+TEST(StreamableGeneratorTest, TestGeneratorAppliesDeflateCompression) {
+  auto generator = generateMultipleElements();
+  generator.setCompressionMethod(
+      ad_utility::content_encoding::CompressionMethod::DEFLATE);
+
+  ASSERT_TRUE(generator.hasNext());
+  auto compressedData = generator.next();
+
+  // compression should cause bytes to drop below the TEST_BUFFER_SIZE
+  // causing all co_yields to be executed without suspension at once
+  // for this test case

```

Comment:
Then also assert, that the  uncompressedData is larger than the testBuffer size,
or that the compressedData is not equal to the input data,
such that the test asserts, that something happens with the data.